contracts/adapters/FlexibleLeverageStrategyAdapter.sol

```diff
@@ -469,6 +612,39 @@ contract FlexibleLeverageStrategyAdapter is BaseAdapter {
         return _calculateCurrentLeverageRatio(currentLeverageInfo.collateralValue, currentLeverageInfo.borrowValue);
     }
 
+    /**
+     * Calculates the total notional rebalance size. Note: this function does not take into account timestamps, so it may return a nonzero value
+     * even when shouldRebalance would return ShouldRebalance.NONE for all exchanges (since minimum delays have not elapsed)
+     *
+     * @return uint256      Total notional rebalance size. Measured in the asset that would be sold
+     * @return bool         Whether it would be levering or delevering
+     */
+    function getTotalRebalanceNotional()  external view returns(uint256, bool) {
+
+        uint256 newLeverageRatio;
+        uint256 currentLeverageRatio = getCurrentLeverageRatio();
+
+        // if over incentivized leverage ratio, always ripcord
+        if (currentLeverageRatio > incentive.incentivizedLeverageRatio) {
+            newLeverageRatio = methodology.maxLeverageRatio;
+        // if we are in an ongoing twap, use the cached twapLeverageRatio as our target leverage
+        } else if (twapLeverageRatio > 0) {
+            newLeverageRatio = twapLeverageRatio;
+        // if all else is false, then we would just use the normal rebalance new leverage ratio calculation
+        } else {
+            newLeverageRatio = _calculateNewLeverageRatio(currentLeverageRatio);
+        }
+
+        ActionInfo memory actionInfo = _createActionInfo();
+
+        bool isLever = newLeverageRatio > currentLeverageRatio;
+        uint256 leverageRatioDifference = isLever ? newLeverageRatio.sub(currentLeverageRatio) : currentLeverageRatio.sub(newLeverageRatio);
+
+        uint256 collateralRebalanceUnits = leverageRatioDifference.preciseDiv(currentLeverageRatio).preciseMul(actionInfo.collateralBalance);
```

Comment:
This should be `notional` not in `units` so think this name is a misnomer

Reply:
Feels like there may be some duplicate logic here and in `_calculateChunkRebalanceNotional `. Which brings up another point do we want to return the`chunkRebalanceNotional` or the `totalRebalanceNotional`?

Reply:
It's way more complicated doing `chunkRebalanceNotional` because we'd need to pass in the `exchangeName` and fetch the max trade size there. Instead, we're planning to have the FLI rebalance viewer logic take care of the chunking logic and passing the chunk notional into the exchange get quote functions

Reply:
I think we can do it either way. It all just depends on whether we want the logic for calculating chunks to go in the viewer contract or here. Any agreement on which way to go with?

Reply:
I think we can do it either way. It all just depends on whether we want the logic for calculating chunks to go in the viewer contract or here. Any agreement on which way to go with?

Reply:
My feeling is that we should pass in an array of exchange names that returns an array of chunk sizes. I think that's a more practical solution and reduces any logic duplication across contracts.